lib/explorer/polars_backend/series.ex

```diff
@@ -682,6 +686,11 @@ defmodule Explorer.PolarsBackend.Series do
 
   defp to_mod_series(value, %{dtype: dtype}, mod),
     do: mod.from_list([value], dtype)
+
+  defp literal!(%Series{} = series), do: series
+
+  defp literal!(value),
+    do: from_list([value], Explorer.Shared.check_types!([value]))
```

Comment:
Similar here. We should use to_mod_series if any of them is a series. For example, `select(bool, float_series, 1)`, then 1 should be a float. Similar thing for binaries/categories/strings. :)